sdk/common/src/main/scala/org/typelevel/otel4s/sdk/Attributes.scala

```diff
@@ -16,49 +16,40 @@
 
 package org.typelevel.otel4s.sdk
 
-import cats.Applicative
 import cats.Hash
-import cats.Monad
 import cats.Monoid
 import cats.Show
 import cats.implicits._
 import org.typelevel.otel4s.Attribute
 import org.typelevel.otel4s.Attribute.KeySelect
 import org.typelevel.otel4s.AttributeKey
 
-/** An immutable collection of [[Attribute]]s.
+/** An immutable collection of [[Attribute]]s. It contains only unique keys.
+  *
+  * @see
+  *   [[https://opentelemetry.io/docs/specs/otel/common/#attribute-collections]]
   */
-final class Attributes private (
-    private val m: Map[AttributeKey[_], Attribute[_]]
-) {
-  def get[T: KeySelect](name: String): Option[Attribute[T]] = {
-    val key = KeySelect[T].make(name)
-    m.get(key).map(_.asInstanceOf[Attribute[T]])
-  }
-  def get[T](key: AttributeKey[T]): Option[Attribute[T]] =
-    m.get(key).map(_.asInstanceOf[Attribute[T]])
-
-  def isEmpty: Boolean = m.isEmpty
-  def size: Int = m.size
-  def contains(key: AttributeKey[_]): Boolean = m.contains(key)
-  def foldLeft[F[_]: Monad, B](z: B)(f: (B, Attribute[_]) => F[B]): F[B] =
-    m.foldLeft(Monad[F].pure(z)) { (acc, v) =>
-      acc.flatMap { b =>
-        f(b, v._2)
-      }
-    }
-  def forall[F[_]: Monad](p: Attribute[_] => F[Boolean]): F[Boolean] =
-    foldLeft(true)((b, a) => {
-      if (b) p(a).map(b && _)
-      else Monad[F].pure(false)
-    })
-  def foreach[F[_]: Applicative](f: Attribute[_] => F[Unit]): F[Unit] =
-    m.foldLeft(Applicative[F].unit) { (acc, v) =>
-      acc *> f(v._2)
-    }
+sealed trait Attributes extends Iterable[Attribute[_]] {
+
+  /** Returns an attribute for the given attribute name, or `None` if not found.
+    */
+  def get[T: KeySelect](name: String): Option[Attribute[T]]
+
+  /** Returns an attribute for the given attribute key, or `None` if not found.
+    */
+  def get[T](key: AttributeKey[T]): Option[Attribute[T]]
 
-  def toMap: Map[AttributeKey[_], Attribute[_]] = m
-  def toList: List[Attribute[_]] = m.values.toList
+  /** Whether this attributes collection contains the given key.
+    */
+  def contains(key: AttributeKey[_]): Boolean
+
+  /** Returns the `Map` representation of the attributes collection.
+    */
+  def toMap: Map[AttributeKey[_], Attribute[_]]
```

Comment:
I'm torn between this being `toMap` vs `asMap`. `asMap` has the implication that `Attributes` is already either like a map (and can be easily wrapped as one) or backed by a map. is that something that we should commit to publicly as its design?

Reply:
Good question. Collections use `to{X}`. Technically, `Attributes` is a collection now. 
Across the code base, we only use `asMap` in `TraceState`.

Reply:
> Collections use `to{X}`

except for conversions (i.e. wrappers) between Java and Scala collections, which use `as{Java,Scala}`